presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileMetadata.java

```diff
@@ -0,0 +1,268 @@
+                Optional.empty(),
+                ImmutableList.of());
+        return ImmutableList.of(new ConnectorTableLayoutResult(layout, constraint.getSummary()));
+    }
+
+    @Override
+    public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTableLayoutHandle handle)
+    {
+        LocalFileTableLayoutHandle layout = checkType(handle, LocalFileTableLayoutHandle.class, "layout");
+        return new ConnectorTableLayout(layout, Optional.empty(), TupleDomain.<ColumnHandle>all(), Optional.empty(), Optional.empty(), ImmutableList.of());
+    }
+
+    @Override
+    public boolean canCreateSampledTables(ConnectorSession session)
+    {
+        return false;
+    }
+
+    @Override
+    public Map<String, ColumnHandle> getColumnHandles(ConnectorSession session, ConnectorTableHandle table)
+    {
+        LocalFileTableHandle tableHandle = checkType(table, LocalFileTableHandle.class, "tableHandle");
+        ImmutableMap.Builder<String, ColumnHandle> columnHandles = ImmutableMap.builder();
+        int index = 0;
+        for (ColumnMetadata column : getColumns(tableHandle)) {
+            int fieldIndex;
+            if (column.equals(SERVER_ADDRESS_COLUMN)) {
+                fieldIndex = -1;
+            }
+            else {
+                fieldIndex = index;
+                index++;
+            }
+            columnHandles.put(column.getName(), new LocalFileColumnHandle(connectorId, column.getName(), column.getType(), fieldIndex));
+        }
+        return columnHandles.build();
+    }
+
+    private List<ColumnMetadata> getColumns(LocalFileTableHandle tableHandle)
+    {
+        checkArgument(tableColumns.containsKey(tableHandle.getSchemaTableName()), "Table %s not registered", tableHandle.getSchemaTableName());
+        return tableColumns.get(tableHandle.getSchemaTableName());
+    }
+
+    @Override
+    public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle)
+    {
+        checkType(tableHandle, LocalFileTableHandle.class, "tableHandle");
+        return checkType(columnHandle, LocalFileColumnHandle.class, "columnHandle").getColumnMetadata();
+    }
+
+    @Override
+    public Map<SchemaTableName, List<ColumnMetadata>> listTableColumns(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        requireNonNull(prefix, "prefix is null");
+        ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> columns = ImmutableMap.builder();
+        for (SchemaTableName tableName : listTables(session, prefix)) {
+            LocalFileTableHandle tableHandle = tables.get(tableName);
+            if (tableHandle != null) {
+                columns.put(tableName, getColumns(tableHandle));
+            }
+        }
+        return columns.build();
+    }
+
+    @Override
+    public void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        throw new UnsupportedOperationException();
```

Comment:
Remove all these methods -- they are default methods in the interface which throw the correct exception type